subsys/bluetooth/host/l2cap.c

```diff
@@ -2952,6 +2952,7 @@ int bt_l2cap_chan_send_cb(struct bt_l2cap_chan *chan, struct net_buf *buf, bt_co
 {
 	struct bt_l2cap_le_chan *le_chan = BT_L2CAP_LE_CHAN(chan);
 	struct l2cap_tx_meta_data *data;
+	void *old_user_data = l2cap_tx_meta_data(buf);
```

Comment:
can you add a check in one of the bsim tests for this? so that it fails without this patch.